python/tests/test_config.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+
+
+from fusion_engine_client.messages.configuration import ConfigurationSource, DeviceCourseOrientationConfig
+from fusion_engine_client.messages import (SetConfigMessage,
+                                           Uart1BaudConfig,
+                                           ConfigType,
+                                           Direction,
+                                           ConfigDataMessage,
+                                           GnssLeverArmConfig,
+                                           InvalidConfig)
+
+
+import logging
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+logging.getLogger('point_one').setLevel(logging.DEBUG)
+
+
+def test_set_config():
+    BASE_SIZE = 8
+
+    set_msg = SetConfigMessage(DeviceCourseOrientationConfig(Direction.BACKWARD, Direction.DOWN))
+    assert len(set_msg.pack()) == BASE_SIZE + 4
+
+    set_msg = SetConfigMessage(GnssLeverArmConfig(1, 2, 3))
+    assert len(set_msg.pack()) == BASE_SIZE + 12
+
+    set_msg = SetConfigMessage()
+    set_msg.config_object = Uart1BaudConfig(9600)
+    uart_data = set_msg.pack()
+    assert len(uart_data) == BASE_SIZE + 4
+
+    set_msg = SetConfigMessage()
+    set_msg.unpack(uart_data)
+    assert isinstance(set_msg.config_object, Uart1BaudConfig)
+    assert set_msg.config_object.GetType() == ConfigType.UART1_BAUD
+    assert set_msg.config_object.value == 9600
+
+
+def test_bad_set_config():
+    BASE_SIZE = 8
+    set_msg = SetConfigMessage()
+    with pytest.raises(TypeError):
+        set_msg.pack()
+
+    set_msg = SetConfigMessage()
+    set_msg.config_object = "Dummy"
+    with pytest.raises(TypeError):
+        set_msg.pack()
+
+    set_msg = SetConfigMessage()
+    set_msg.config_object = InvalidConfig()
+    uart_data = set_msg.pack()
```

Comment:
Why doesn't this raise?

Reply:
I specifically defined InvalidConfig to be able to be packed/unpacked without an exception. InvalidConfig is one of the generated `ConfigClass` . It's up to the user to handle it.